### DISCUSSION TITLE
Список hash экземпляров временной таблицы

### DISCUSSION THREAD

**Discussion by juramebel** [totumonline/totum-ru-issues-and-discussions#431] [Как это сделать?] [answered]
Добрый день!
Необходимо сделать автообновление временных таблиц, используемых в качестве рабочих мест. Пока идея сделать это через обновление h_insert всех живых экземпляров, но нигде не могу найти, где взять список их hash. 
Сможете подсказать?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Получить системно из БД — нет такой возможности. Можно как-нибудь кодом действия изнутри таблиц. Хэш получается переменной $#nth.

**Reply by juramebel** [reply]
Тогда можно ли осуществить такой вариант? 
При создании временной таблицы записывается ее хэш в вспомогательную. 
Кроном проверяется не истекло ли время жизни, и если да - запись удаляется. 
Обновления проводятся используя данные этой вспомогательной таблицы.
Только не могу нигде найти, от чего зависит время жизни временной таблицы. 
В документации указано, что обычно 3 часа, а что на это влияет - не нашел.

**Comment by juramebel** [comment]
В общем вопрос закрыт. 
Решил через вспомогательную таблицу - реестр всех временных таблиц. 
И проверкой периодической с помощью функции tryCatch их существования.

**Comment by poluninatm** [comment]
Временная таблица доступна только пользователю, от которого она была
создана. Проверяйте вашу логику, учитывая этот момент.

Ср, 23 авг. 2023 г. в 08:01, juramebel ***@***.***>:

**Reply by juramebel** [reply]
Хм... интересный момент. Соберу тестовый стенд с кроном, проверю. 
Спасибо!

**Comment by poluninatm** [comment]
В качестве личного кабинета задумывалась таблица циклов с одним циклом на
пользователя- там есть настройка. Таблицы внутри циклов могут быть доступны
разные в зависимости от ролей. Посмотрите в эту сторону - возможно, там
хватит инструментов.

Ср, 23 авг. 2023 г. в 08:57, juramebel ***@***.***>:

**Reply by juramebel** [reply]
Это вот эта настройка?
Честно говоря не знал. 
 Возможно многое бы сделал иначе уже)

![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/19141501/07b17970-11e7-44de-990d-208105525e66)

**Reply by juramebel** [reply]
Попробовал. Отличная штука)
Жаль раньше не знал.

**Reply by ttm-alexeypolunin** [reply]
Обязательно поставьте таблице циклов тип доступа Видит только свои циклы, тк иначе методом подбора номеров других циклов можно получить к ним доступ несмотря на то, что таблица циклов недоступна к показу.
